e2e/compatibility/terraform_download_test.go

```diff
@@ -0,0 +1,65 @@
+//go:build e2e
+// +build e2e
+
+package compatibility
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/consul-terraform-sync/config"
+	"github.com/hashicorp/consul-terraform-sync/driver"
+	"github.com/hashicorp/consul-terraform-sync/testutils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// This technically isn't an e2e test, but it downloads a file from the internet,
+// so it likely shouldn't be run frequently with other tests.
+func TestCompatibility_TFDownload(t *testing.T) {
+
+	tempDir := fmt.Sprintf("%s%s", tempDirPrefix, "Compatibility_TFDownload")
+	cleanup := testutils.MakeTempDir(t, tempDir)
+	defer cleanup()
+
+	conf := config.DefaultTerraformConfig()
+	conf.Path = &tempDir
+	// Use blank as a default value. Nil results in a panic.
+	tfVersion := ""
+	conf.Version = &tfVersion
+
+	binaryPath := path.Join(tempDir, "terraform")
+	info, err := os.Stat(binaryPath)
+	// We want to ensure the file does not exist and remove it if found.
+	if !os.IsNotExist(err) {
+		require.NoError(t, err, "An unexpected error occurred while checking for old terraform executable.")
+	}
+	if info != nil {
+		err = os.Remove(binaryPath)
+		require.NoError(t, err, "An unexpected error occurred while removing an old terraform executable.")
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	err = driver.InstallTerraform(ctx, conf)
+	require.NoError(t, err, "Unable to download terraform executable.")
+
+	// Check to ensure the terraform binary exists.
+	info, err = os.Stat(binaryPath)
+	require.NoError(t, err, "Unable to find terraform executable that was expected to be installed.")
+	// Check for an arbitrary 1MB+ terraform executable size.
+	assert.Greater(t, info.Size(), int64(1_000_000), "The terraform file size was unexpectedly small. This is likely an issue with installing.")
+
+	// Trigger install again. It shouldn't re-download the file.
+	firstInstallTime := info.ModTime()
+	ctx, cancel = context.WithTimeout(context.Background(), 3*time.Second)
```

Comment:
Is this too aggressive of a timeout? in case the CPU is busy for instance

Reply:
I suppose it could be. It's hitting the local disk, so I figured 3s was enough. I can bump it to something higher if you want, though.